### ISSUE TITLE
[BUG] Performance issues when schema has data Buffer type

### ISSUE THREAD

**Issue by bushev** [dynamoose/dynamoose#1205]
### Summary:

The DDB records have a binary data size of 200KB approximately. When I use Dynamoose the scan took 18 seconds, when, I use AWS DDB client, it takes ~100ms. 

I suspect that Dynamoose has some performance issues in a Buffer type serialization. When I remove a field with Buffer type from the schema, it starts working fast as expected.

### Code sample:

```typescript
console.time('scan');
const result = await myModel.scan().exec();
console.timeEnd('scan');
console.log(result);
```

#### Schema
```
const mySchema = {
  site: {
    type: Object,
    required: true,
    schema: {
      pages: {
        type: Buffer,
        required: true
      },
      ...
    }
  }
}
```

#### Model

```
const DEFAULT_OPTIONS = {create: false, waitForActive: {enabled: false}};

const myModel: ModelType<MyEntity> & IMyModelMethods = dynamoose.model<MyEntity>(
  'MyTable',
  mySchema,
  DEFAULT_OPTIONS
) as ModelType<MyEntity> & IMyModelMethods;
```

### Current output and behavior (including stack trace):

```
scan: 19.177s // <---- That's the issue

{
...
count: 53,
scannedCount: 53,
timesScanned: 1,
}
```

### Expected output and behavior:

The scan should take much less than 19 seconds.

### Environment:

Operating System: MacOS
Operating System Version: 11.3.1
Node.js version (`node -v`): v14.16.0
NPM version: (`npm -v`): 6.14.11
Dynamoose version: 2.7.3

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
@bushev That isn't good.

Any chance you can make a reproducible example of this? It should look something like the following:

```js
const dynamoose = require("dynamoose");

(async () => {
    dynamoose.aws.ddb.local();

    const schema = new dynamoose.Schema({
        // CODE HERE
    });

    const Model = dynamoose.model(`Model-${Date.now()}`, schema);

    await Model.create({
        // CODE HERE
    });

    console.time('scan');
    const result = await myModel.scan().exec();
    console.timeEnd('scan');
    console.log(result);

    console.time('scanaws');
    const resultB = await dynamoose.aws.ddb().scan({
        // CODE HERE
    }).promise();
    console.timeEnd('scanaws');
    console.log(resultB);
})();
```

Hopefully I got all of that right (you will need to verify that I don't have any typos or bugs in it). But the goal is for me to follow the steps to reproduce:

1. Copy that code into a new file
2. `npm i dynamoose`
3. Start local DynamoDB instance
4. Run file
5. See the issue

Currently with your example we are missing a few pieces (like what the buffer is, how many items are in your table, etc).

So really trying to get a full self contained example here.

**Comment by bushev** [comment]
Hey @fishcharlie, thanks for the quick reply!

Im running this code:

```javascript
const dynamoose = require('dynamoose');

const tableName = `Model-${Date.now()}`;

process.env.AWS_REGION = 'us-east-1';

(async () => {
  try {
    dynamoose.aws.ddb.local('http://127.0.0.1:7011');

    const schema = new dynamoose.Schema({
      myData: {
        type: Buffer
      }
    });

    const myModel = dynamoose.model(tableName, schema, {
      create: true,
      waitForActive: {
        enabled: true
      }
    });

    console.time('create');
    await myModel.create({
      myData: Buffer.alloc(1024 * 200).fill(null)
    });

    console.time('scan');
    const result = await myModel.scan().exec();
    console.timeEnd('scan');
    console.log(result);

    console.time('scan_aws');
    const resultB = await dynamoose.aws
      .ddb()
      .scan({
        TableName: tableName
      })
      .promise();
    console.timeEnd('scan_aws');
    console.log(resultB);
  } catch (err) {
    console.timeEnd('create');
    console.log(err.message);
  }
})();
```

I'm not sure, why it didn't connect to my local DDB on 7011 port and trying to connect to AWS instead. But it also shows 2 seconds delay and "Maximum call stack size exceeded" error only in the case of a 200K buffer.

Results are:

![image](https://user-images.githubusercontent.com/7535766/118307477-c2327180-b4b8-11eb-95a4-1ef6218b2a5c.png)

**Comment by fishcharlie** [comment]
@bushev That is helpful. I'll try to run this sometime soon. Sadly I'm going out of town and won't be back until the 24th. I'll see if I can carve out sometime during that trip to look into this more, but no guarantees on that front.

Sorry you are running into this, and hopefully I'll be able to look into this more soon.

**Comment by bushev** [comment]
Hey @fishcharlie, any update regarding the issues?

**Comment by kishorechavali** [comment]
We are also facing this issue when dealing with Buffer types , following code is seeing buffer type as object and trying to loop through all the entries for attribute types and resulting in above error , instead of typeof may be we should use instanceof ? here 

    const isObject = typeof value === "object" && value !== null;

**Comment by kishorechavali** [comment]
@fishcharlie create PR https://github.com/dynamoose/dynamoose/pull/1225 can you please review

**Comment by fishcharlie** [comment]
Fixed by #1225
